### ISSUE TITLE
Implement OSC Validator

### ISSUE THREAD

**Issue by kewitz** [opencollective/opencollective#5974]
- [ ] Create OSC validator function that is called every time a collective applies to OSC
- [ ] Has default behavior for insufficient data (sends to manual approval?)
- [ ] Implement conditionals for minimum criteria
    **? What are the parameters for the minimum criteria validator?**
    - [ ] Last Activity
    - [ ] Has Number of Contributors
    - [ ] Belongs to set of Licenses
    - [ ] Is Fork

**Comment by iamronen** [comment]
attn @poohlaga and @RichardLitt

**Comment by BenJam** [comment]
> Has default behavior for insufficient data (sends to manual approval?)

++m on this, the idea is to answer these questions if we can and back down to the form (with elements we _can_ answer pre-filled) if we cant.

**Comment by znarf** [comment]
- [ ] Last activity -> can be fetched with GitHub GraphQL (1)
- [ ] Number of contributors -> can be fetched with GitHub GraphQL (2)
- [ ] SPDX compliant license -> can be easily added as `license` with current code
- [ ] Is not a fork -> already available as `fork` (`true` or `false`)
- [ ] Is owned by an organization -> already available in `owner.type`

(1) 

```graphql
defaultBranchRef {
  target {
    ... on Commit {
      committedDate
    }
  }
}
```

(2)

```graphql
collaborators {
  totalCount
}
```

But GraphQL doesn't allow something like we have today (https://docs.github.com/en/rest/repos/repos#list-repositories-for-the-authenticated-user).

**Comment by kewitz** [comment]
@BenJam @RichardLitt can more than one collective be created using the same connected repository?

**Comment by RichardLitt** [comment]
Yes, but probably not ideal to allow that for the automatic validator.

**Comment by BenJam** [comment]
@RichardLitt I think that might be counter-productive for users, we should alert (and direct the user) to the collective if one already exists for that organisation or repo.

**Comment by RichardLitt** [comment]
Thank you, @gustavlrsn!
